### ISSUE TITLE
Python broken with version 3.8.4 and Flask_SQLAlchemy

### ISSUE THREAD

**Issue by cleberjsantos** [docker-library/python#506]
**Bug** https://bugs.python.org/issue39960

## Resume

This bug are affecting some projects based on latest version python alpine (Python 3.8.4)

### Sample traceback.

```Python
    app = create_app()
  File "/restapi/restapi/__init__.py", line 21, in create_app
    from . import extensions
  File "/restapi/restapi/extensions/__init__.py", line 22, in <module>
    db = SQLAlchemy(engine_options={'poolclass': NullPool})
  File "/usr/local/lib/python3.8/site-packages/Flask_SQLAlchemy-2.4.1-py3.8.egg/flask_sqlalchemy/__init__.py", line 716, in __init__
    self.Model = self.make_declarative_base(model_class, metadata)
  File "/usr/local/lib/python3.8/site-packages/Flask_SQLAlchemy-2.4.1-py3.8.egg/flask_sqlalchemy/__init__.py", line 798, in make_declarative_base
    model.query_class = self.Query
  File "/usr/local/lib/python3.8/site-packages/SQLAlchemy-1.3.17-py3.8-linux-x86_64.egg/sqlalchemy/ext/declarative/api.py", line 79, in __setattr__
    _add_attribute(cls, key, value)
  File "/usr/local/lib/python3.8/site-packages/SQLAlchemy-1.3.17-py3.8-linux-x86_64.egg/sqlalchemy/ext/declarative/base.py", line 802, in _add_attribute
    type.__setattr__(cls, key, value)
TypeError: can't apply this __setattr__ to DefaultMeta object
```

**Comment by tianon** [comment]
It's unfortunate that this has happened, but I'm not sure I understand what there is for *us* to do about it? (It's a bug in Python itself, which we package/provide faithfully unchanged, so that upstream discussion is the appropriate place to find a resolution to this. :confused:)

**Comment by cleberjsantos** [comment]
@tianon What if an Alpine image earlier version is given as a help?
In this way, we can use version 3.8.3 and not break projects already based on Alpine version 3.8.

**Comment by tianon** [comment]
The older tags are still available, they just won't ever get updated: https://hub.docker.com/_/python?tab=tags&name=3.8.3-alpine

If this is a "showstopper" bug for using Python 3.8 on Alpine and upstream isn't interested in fixing it, we'd be more likely to drop support for Alpine entirely (but the upstream discussion looks favorable to finding a proper fix, and it doesn't seem like this issue affects *all* uses of Python 3.8 on Alpine, just certain use cases).

**Comment by cleberjsantos** [comment]
I believe that if there are only a few use cases with problems, we can wait, this discussion's good for those who are already using the base version 3.8 of Alpine, to have an alternative until the definitive solution is launched or a patch version.

**Comment by decaz** [comment]
@tianon Python 3.8.5 was released and I think this regression (https://bugs.python.org/issue41295) was resolved there: https://docs.python.org/release/3.8.5/whatsnew/changelog.html#python-3-8-5-final.

Please, update Python version so new image is being built.

**Comment by yosifkit** [comment]
Fixed by `3.8.5` (https://github.com/docker-library/python/commit/44ded7c92b0c20abbeaa88785751eb0822d05ced)

[Readme](https://github.com/docker-library/python#see-a-change-merged-here-that-doesnt-show-up-on-docker-hub-yet) on updates and so, https://github.com/docker-library/official-images/pull/8397.
